src/components/ManagerPage/FirmwareUpdate.js

```diff
@@ -81,18 +81,19 @@ class FirmwareUpdate extends PureComponent<Props, State> {
   render() {
     const { deviceInfo, t, device } = this.props
     const { firmware, modal, stepId, ready } = this.state
+
+    const deviceSpecs = getDeviceModel(device.modelId)
+
     return (
       <Card p={4}>
         <Box horizontal align="center" flow={2}>
           <Box color="dark">
-            {device.product === 'Blue' ? <Blue size={30} /> : <NanoS size={30} />}
+            {deviceSpecs.id === 'blue' ? <Blue size={30} /> : <NanoS size={30} />}
```

Comment:
note for future refactoring: we should have a component map like this: `{ blue: Blue, nanoS: NanoS }`

Reply:
Yup I was thinking about it as well. Will come up with something like this when implementing Nano X designs